contracts/exchange/contracts/src/MixinSignatureValidator.sol

```diff
@@ -88,19 +95,138 @@ contract MixinSignatureValidator is
         );
     }
 
+    /// @dev Approves/unnapproves an OrderValidator contract to verify signatures on signer's behalf
+    ///      using the `OrderValidator` signature type.
+    /// @param validatorAddress Address of Validator contract.
+    /// @param approval Approval or disapproval of  Validator contract.
+    function setOrderValidatorApproval(
+        address validatorAddress,
+        bool approval
+    )
+        external
+        nonReentrant
+    {
+        address signerAddress = getCurrentContextAddress();
+        allowedOrderValidators[signerAddress][validatorAddress] = approval;
+        emit SignatureValidatorApproval(
+            signerAddress,
+            validatorAddress,
+            approval
+        );
+    }
+
+    /// @dev Verifies that a signature for an order is valid.
+    /// @param order The order.
+    /// @param signerAddress Address that should have signed the given order.
+    /// @param signature Proof that the order has been signed by signer.
+    /// @return True if the signature is valid for the given order and signer.
+    function isValidOrderSignature(
+        Order memory order,
+        address signerAddress,
+        bytes memory signature
+    )
+        public
+        view
+        returns (bool isValid)
+    {
+        bytes32 orderHash = getOrderHash(order);
+        return isValidOrderWithHashSignature(
+            order,
+            orderHash,
+            signerAddress,
+            signature
+        );
+    }
+
     /// @dev Verifies that a hash has been signed by the given signer.
-    /// @param hash Any 32 byte hash.
-    /// @param signerAddress Address that should have signed the given hash.
+    /// @param hash Any 32-byte hash.
+    /// @param signerAddress Address that should have signed the.Signat given hash.
     /// @param signature Proof that the hash has been signed by signer.
-    /// @return True if the address recovered from the provided signature matches the input signer address.
-    function isValidSignature(
+    /// @return True if the signature is valid for the given hash and signer.
+    function isValidHashSignature(
         bytes32 hash,
         address signerAddress,
         bytes memory signature
     )
         public
         view
         returns (bool isValid)
+    {
+        SignatureType signatureType = readValidSignatureType(hash, signerAddress, signature);
```

Comment:
Nit: I'd default to putting any function calls with >2 arguments onto multiple lines.